rows.go

```diff
@@ -127,6 +127,8 @@ func (r *rows) ColumnTypeScanType(index int) reflect.Type {
 		return reflect.TypeOf(map[string]any{})
 	case TYPE_MAP:
 		return reflect.TypeOf(Map{})
+	case TYPE_ARRAY:
+		return reflect.TypeOf([]any{})
```

Comment:
Is there any better way to reflect the `ARRAY` type here?

Reply:
Maybe sth. like:

```
reflect.TypeOf([0]any{})
```

Otherwise it would be a slice I think.

Reply:
We are returning a slice because we cannot dynamically return differently sized arrays, to the best of my knowledge... Scanning into the composite array type works, but I am not sure how I'd generalize that to return an array in `vector_getters.`